nginx/Dockerfile-alpine

```diff
@@ -9,40 +87,132 @@ RUN set -eux; \
+    PROXY_SSL_CIPHERS="ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384" \
+    PROXY_SSL_PREFER_CIPHERS=off \
+    PROXY_SSL_VERIFY=off \
+    PROXY_SSL_OCSP_STAPLING=off \
+    SERVER_NAME=localhost \
+    SSL_PORT=443 \
+    TIMEOUT=60s \
+    WORKER_CONNECTIONS=1024 \
+    LD_LIBRARY_PATH=/lib:/usr/lib:/usr/local/lib \
+    NGINX_ENVSUBST_OUTPUT_DIR=/etc/nginx \
+    # overridden variables
+    USER=nginx \
+    # CRS specific variables
     PARANOIA=1 \
     ANOMALY_INBOUND=5 \
     ANOMALY_OUTBOUND=4 \
     BLOCKING_PARANOIA=1
 
+COPY --from=build /usr/local/modsecurity/lib/libmodsecurity.so.${MODSEC_VERSION} /usr/local/modsecurity/lib/
+COPY --from=build /etc/nginx/modules/ngx_http_modsecurity_module.so /etc/nginx/modules/ngx_http_modsecurity_module.so
+COPY --from=build /usr/share/TLS/server.key /etc/nginx/conf/server.key
+COPY --from=build /usr/share/TLS/server.crt /etc/nginx/conf/server.crt
+COPY --from=build /usr/share/TLS/dhparam-* /etc/ssl/certs/
+COPY --from=build /etc/modsecurity.d/unicode.mapping /etc/modsecurity.d/unicode.mapping
+COPY --from=build /etc/modsecurity.d/modsecurity.conf /etc/modsecurity.d/modsecurity.conf
+COPY --from=crs_release /opt/owasp-crs /opt/owasp-crs
 # We use the templating mechanism from the nginx image here,
 # as set up by owasp/modsecurity-docker
 COPY nginx/templates /etc/nginx/templates/
+COPY src/etc/modsecurity.d/modsecurity-override.conf /etc/nginx/templates/modsecurity.d/modsecurity-override.conf.template
 COPY src/etc/modsecurity.d/setup.conf /etc/nginx/templates/modsecurity.d/setup.conf.template
+COPY src/bin/healthcheck /usr/local/bin/healthcheck
+COPY nginx/docker-entrypoint.d/*.sh /docker-entrypoint.d/
 COPY src/opt/modsecurity/activate-rules.sh /docker-entrypoint.d/95-activate-rules.sh
-COPY --from=release /opt/owasp-crs /opt/owasp-crs
 
-# Alpine needs GNU 'sed' because the 'sed' program shipped with busybox does not support 'z' parameter for separating lines with a 'NUL' character.
+
 RUN set -eux; \
     apk add --no-cache \
-    sed; \
-    ln -sv /opt/owasp-crs /etc/modsecurity.d/
+        curl \
+        curl-dev \
+        libfuzzy2 \
+        libmaxminddb-dev \
+        libstdc++ \
+        libxml2-dev \
+        lmdb-dev \
+        moreutils \
+        tzdata \
+        pcre \
+        pcre2 \
+        # Alpine needs GNU 'sed' because the 'sed' program shipped with busybox does not support 'z' parameter for separating lines with a 'NUL' character.
+        sed \
+        yajl; \
+    mkdir /etc/nginx/ssl/; \
+    mkdir -p /tmp/modsecurity/data; \
+    mkdir -p /tmp/modsecurity/upload; \
+    mkdir -p /tmp/modsecurity/tmp; \
+    chown -R nginx:nginx /usr/share/nginx /tmp/modsecurity; \
+    # Comment out the SecDisableBackendCompression option since it is not supported in V3
```

Comment:
Put the comment at the top or end it with `\`, otherwise the command ends on this line.

Reply:
Example:
```
❯ cat Dockerfile
FROM alpine

RUN apk add sed \
    #THIS IS A COMMENT
    ca-certificates \
    #THIS IS ANOTHER COMMENT
    vim;


❯ docker build -t ll .
[+] Building 9.7s (7/7) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                       0.0s
 => => transferring dockerfile: 159B                                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                            0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                           2.6s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                              0.0s
 => [1/2] FROM docker.io/library/alpine@sha256:b95359c2505145f16c6aa384f9cc74eeff78eb36d308ca4fd902eeeb0a0b161b                                                            0.0s
 => => resolve docker.io/library/alpine@sha256:b95359c2505145f16c6aa384f9cc74eeff78eb36d308ca4fd902eeeb0a0b161b                                                            0.0s
 => => sha256:b95359c2505145f16c6aa384f9cc74eeff78eb36d308ca4fd902eeeb0a0b161b 1.64kB / 1.64kB                                                                             0.0s
 => => sha256:3d426b0bfc361d6e8303f51459f17782b219dece42a1c7fe463b6014b189c86d 528B / 528B                                                                                 0.0s
 => => sha256:bfe296a525011f7eb76075d688c681ca4feaad5afe3b142b36e30f1a171dc99a 1.47kB / 1.47kB                                                                             0.0s
 => [2/2] RUN apk add sed     ca-certificates     vim;                                                                                                                     6.6s
 => exporting to image                                                                                                                                                     0.3s
 => => exporting layers                                                                                                                                                    0.3s
 => => writing image sha256:766811ce46f12b980b6289d5307f84b9eed844c5a7f6e6246c30a19e3affe194                                                                               0.0s
 => => naming to docker.io/library/ll
```

Here you can see that the build is running `RUN apk add sed     ca-certificates     vim;`, which is the expected result.

Reply:
Interesting. That seems to be shell dependent. In my zsh the comment line terminates the command.